### PR TITLE
fixes text overflow issue  fixes #251

### LINKED ISSUE #251: leader board page's leader card  overlaps in md and lg screen
Screenshot 2024-02-25 at 2 27 42 AM

### PR DESCRIPTION
**Pull request by rhithesh**
fixes #251

### CODE REVIEW

File: package-lock.json
Comment (rithviknishad) [inline package-lock.json:1]: Remove `package-lock.json` file

File: components/contributors/LeaderboardCard.tsx [hunk 1]
```diff
@@ -64,12 +64,14 @@ export default function LeaderboardCard({
                   alt={contributor.user.social.github}
                 />
               </div>
-              <div className="ml-4">
-                <div className="font-bold text-green-500 truncate">
+              <div className="ml-4 mr-4  pr-10  text-wrap basis-[60%] ">
```
Comment (rithviknishad) [inline components/contributors/LeaderboardCard.tsx:67]:
```suggestion
              <div className="ml-4 mr-4 pr-10 text-wrap basis-[60%]">
```

File: components/contributors/LeaderboardCard.tsx [hunk 2]
```diff
@@ -36,17 +36,17 @@ export default function LeaderboardCard({
...
           >
             {position + 1}
           </div>
         )}
-        <div className="flex md:flex-row flex-col md:items-center justify-between w-full py-4">
-          <div className="flex w-full">
-            <div className="min-w-0 flex-1 flex items-center">
+        <div className="flex md:flex-row flex-col md:items-center justify-between w-full py-4 ">
+          <div className="flex w-full   ">
+            <div className="min-w-0 flex-1 flex items-center ">
```
Comment (rithviknishad) [inline components/contributors/LeaderboardCard.tsx:49]:
```suggestion
            <div className="min-w-0 flex-1 flex items-center">
```

File: components/contributors/LeaderboardCard.tsx [hunk 3]
```diff
@@ -36,17 +36,17 @@ export default function LeaderboardCard({
...
             className={`w-10 h-10 rounded-full flex items-center justify-center bg-gradient-to-tr text-white ${badgeColors} border-4 mr-4 shrink-0`}
           >
             {position + 1}
           </div>
         )}
-        <div className="flex md:flex-row flex-col md:items-center justify-between w-full py-4">
-          <div className="flex w-full">
-            <div className="min-w-0 flex-1 flex items-center">
+        <div className="flex md:flex-row flex-col md:items-center justify-between w-full py-4 ">
+          <div className="flex w-full   ">
```
Comment (rithviknishad) [inline components/contributors/LeaderboardCard.tsx:48]:
```suggestion
          <div className="flex w-full">
```

File: components/contributors/LeaderboardCard.tsx [hunk 4]
```diff
@@ -36,17 +36,17 @@ export default function LeaderboardCard({
       href={"/contributors/" + contributor.user.social.github}
       className="block"
     >
-      <div className="flex border-2 border-transparent hover:border-primary-400 hover:shadow-lg hover:scale-105 transition-all duration-300 ease-in-out transform rounded-lg p-4 md:items-center px-2 sm:px-6 md:py-0 py-2 cursor-pointer">
+      <div className="flex border-2 border-transparent hover:border-primary-400 hover:shadow-lg hover:scale-105 transition-all duration-300 ease-in-out transform rounded-lg p-4 md:items-center px-2 sm:px-6 md:py-0 py-2 cursor-pointer   space-y-4    ">
```
Comment (rithviknishad) [inline components/contributors/LeaderboardCard.tsx:39]:
```suggestion
      <div className="flex border-2 border-transparent hover:border-primary-400 hover:shadow-lg hover:scale-105 transition-all duration-300 ease-in-out transform rounded-lg p-4 md:items-center px-2 sm:px-6 md:py-0 py-2 cursor-pointer space-y-4">
```

File: app/leaderboard/Leaderboard.tsx [hunk 5]
```diff
@@ -82,7 +82,7 @@ export default function Leaderboard(props: { data: LeaderboardAPIResponse }) {
           <div className="lg:grid lg:grid-cols-12 2xl:gap-5 px-0 pb-10 lg:pb-20">
             <div className="lg:col-span-7 2xl:col-span-8">
               <div className="sticky top-0 pt-6">
-                <div className="terminal-container-bg border rounded-lg border-primary-500">
+                <div className="terminal-container-bg border rounded-lg border-primary-500 ">
```
Comment (rithviknishad) [inline app/leaderboard/Leaderboard.tsx:85]:
```suggestion
                <div className="terminal-container-bg border rounded-lg border-primary-500">
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (rhithesh) [reply package-lock.json:1]: done!

**Comment by rhithesh** [comment]
@rithviknishad  hi please look into the pr thanks

**Review by rithviknishad — APPROVED** [review]

File: components/contributors/LeaderboardCard.tsx [hunk 6]
```diff
@@ -64,12 +64,14 @@ export default function LeaderboardCard({
                   alt={contributor.user.social.github}
                 />
               </div>
-              <div className="ml-4">
-                <div className="font-bold text-green-500 truncate">
+              <div className="ml-4 mr-4 pr-10 text-wrap basis-[60%]">
+                <div className="font-bold text-green-500 truncate  border w-[180px] ">
```
Comment (rithviknishad) [inline components/contributors/LeaderboardCard.tsx:68]:
```suggestion
                <div className="font-bold text-green-500 truncate w-[180px]">
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]
It does not look good in mobile view

image

**Comment by rhithesh** [comment]
Screenshot 2024-02-26 at 7 02 26 PM
is this fine @rithviknishad ?

File: components/contributors/LeaderboardCard.tsx [hunk 7]
```diff
@@ -36,26 +36,26 @@ export default function LeaderboardCard({
       href={"/contributors/" + contributor.user.social.github}
       className="block"
     >
-      <div className="flex border-2 border-transparent hover:border-primary-400 hover:shadow-lg hover:scale-105 transition-all duration-300 ease-in-out transform rounded-lg p-4 md:items-center px-2 sm:px-6 md:py-0 py-2 cursor-pointer">
+      <div className="flex border-2 border-transparent hover:border-primary-400 hover:shadow-lg hover:scale-105 transition-all duration-300 ease-in-out transform rounded-lg p-4 md:items-center px-2 sm:px-6 md:py-0 py-2 cursor-pointer space-y-4 justify-center  ">
```
Comment (rithviknishad) [inline components/contributors/LeaderboardCard.tsx:39]:
```suggestion
      <div className="flex border-2 border-transparent hover:border-primary-400 hover:shadow-lg hover:scale-105 transition-all duration-300 ease-in-out transform rounded-lg p-4 md:items-center px-2 sm:px-6 md:py-0 py-2 cursor-pointer space-y-4 justify-center">
```

File: components/contributors/LeaderboardCard.tsx [hunk 8]
```diff
@@ -36,26 +36,26 @@ export default function LeaderboardCard({
...
         {!hideBadges && (
           <div
-            className={`w-10 h-10 rounded-full flex items-center justify-center bg-gradient-to-tr text-white ${badgeColors} border-4 mr-4 shrink-0`}
+            className={` h-10  my-6  ml-3 rounded-full w-[40px]   flex items-center justify-center bg-gradient-to-tr text-white ${badgeColors} border-4 mr-4 shrink-0`}
```
Comment (rithviknishad) [inline components/contributors/LeaderboardCard.tsx:42]:
```suggestion
            className={`h-10 my-6 ml-3 rounded-full w-[40px] flex items-center justify-center bg-gradient-to-tr text-white ${badgeColors} border-4 mr-4 shrink-0`}
```

File: components/contributors/LeaderboardCard.tsx [hunk 9]
```diff
@@ -36,26 +36,26 @@ export default function LeaderboardCard({
...
-                className={`flex-shrink-0 ring-2 dark:ring-1 ring-current rounded-full ${
+                className={`flex-shrink-0 ring-2 dark:ring-1 ring-current rounded-full  ${
                   ["text-yellow-600", "text-stone-600", "text-amber-700"][
                     position
                   ] ?? "text-purple-600"
                 }`}
               >
                 <Image
-                  className={`h-14 w-14 rounded-full ${
+                  className={`h-14 w-14 rounded-full   ${
```
Comment (rithviknishad) [inline components/contributors/LeaderboardCard.tsx:58]:
```suggestion
                  className={`h-14 w-14 rounded-full ${
```

File: components/contributors/LeaderboardCard.tsx [hunk 10]
```diff
@@ -64,19 +64,21 @@ export default function LeaderboardCard({
...
+                  </span>
                 </p>
               </div>
             </div>
           </div>
 
-          <div className="px-4 flex items-center justify-between">
-            <div className="md:block">
+          <div className="px-4 flex items-center justify-between  ">
+            <div className="md:block mt-4 md:mt-0   ">
```
Comment (rithviknishad) [inline components/contributors/LeaderboardCard.tsx:81]:
```suggestion
            <div className="md:block mt-4 md:mt-0">
```

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**